src/opossum_lib/core/entities/opossum.py

```diff
@@ -0,0 +1,26 @@
+# SPDX-FileCopyrightText: TNG Technology Consulting GmbH <https://www.tngtech.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from pydantic import BaseModel, ConfigDict
+
+from opossum_lib.core.entities.scan_results import ScanResults
+from opossum_lib.shared.entities.opossum_file_model import OpossumFileModel
+from opossum_lib.shared.entities.opossum_output_file_model import OpossumOutputFileModel
+
+type OpossumPackageIdentifier = str
+type ResourcePath = str
+
+
+class Opossum(BaseModel):
+    model_config = ConfigDict(frozen=True, extra="forbid")
+    scan_results: ScanResults
+    review_results: OpossumOutputFileModel | None = None
+
+    def to_opossum_model(self) -> OpossumFileModel:
```

Comment:
When reading the tests I noticed that I didn't immediately understand what `to_opossum_model` does. I think we should rename to `to_opossum_file_model` or maybe just `to_file_model` since the "opossum" part is kinda clear.